joint_trajectory_controller/src/trajectory.cpp

```diff
@@ -23,20 +23,36 @@
 #include "std_msgs/msg/header.hpp"
 namespace joint_trajectory_controller
 {
-Trajectory::Trajectory() : trajectory_start_time_(0), time_before_traj_msg_(0) {}
 
-Trajectory::Trajectory(std::shared_ptr<trajectory_msgs::msg::JointTrajectory> joint_trajectory)
-: trajectory_msg_(joint_trajectory),
-  trajectory_start_time_(static_cast<rclcpp::Time>(joint_trajectory->header.stamp))
+namespace
 {
+// Safe default joint kinematic limits for Ruckig, in case none is defined in the URDF or
+// joint_limits.yaml
+constexpr double DEFAULT_MAX_VELOCITY = 5;       // rad/s
+constexpr double DEFAULT_MAX_ACCELERATION = 10;  // rad/s^2
+constexpr double DEFAULT_MAX_JERK = 1000;        // rad/s^3
 }
 
+Trajectory::Trajectory()
+: trajectory_start_time_(0),
+  time_before_traj_msg_(0),
+  sampled_already_(false),
+  have_previous_ruckig_output_(false) {}
+
+Trajectory::Trajectory(std::shared_ptr<trajectory_msgs::msg::JointTrajectory> joint_trajectory)
+: trajectory_msg_(joint_trajectory),
+  trajectory_start_time_(static_cast<rclcpp::Time>(joint_trajectory->header.stamp)),
+  sampled_already_(false),
+  have_previous_ruckig_output_(false) {}
+
```

Comment:
Why those two additional constructors?

Reply:
Well, it's the same two constructors. I just initialized a couple more variables in the initializer list. `sampled_already_` and `have_previous_ruckig_output_` were initialized.